pkg/eval/builtin_special.go

```diff
@@ -41,8 +41,17 @@ var IsBuiltinSpecial = map[string]bool{}
 // NoSuchModule encodes an error where a module spec cannot be resolved.
 type NoSuchModule struct{ spec string }
 
+// ModuleLoadError encapsulates an error message a failed module load operation.
+type ModuleLoadError struct {
+	spec     string
+	errormsg string
+}
+
 // Error implements the error interface.
 func (err NoSuchModule) Error() string { return "no such module: " + err.spec }
+func (err ModuleLoadError) Error() string {
+	return "when trying to load module: \"" + err.spec + "\" this error occurred: " + err.errormsg
```

Comment:
Elvish's codebase follows this convention: https://go.dev/wiki/CodeReviewComments#error-strings. I'd write this error message as simply:

```go
"load module " + err.spec + ": " + err.errmsg
```